### ISSUE TITLE
[New Feature] Make SilentLogger class public

### ISSUE THREAD

**Issue by FaustoNascimento** [serilog/serilog#1209]
I quite like the concept of the static Log class which makes usage of the SilentLogger and is a very simple yet effective ILogger to have access to and I can see various valid use cases for it, but it is defined as internal.

My personal use case for it right now is that I would like to extend what the static class Log does and bearing in mind extension methods cannot be applied to static classes (yet?) I need to create a new static class altogether.

It's a very simple change that I think makes sense to have.

**Comment by nblumhardt** [comment]
Thanks for the note.

Would wrapping the `Log` class with your own static class (i.e.  passing through the logger instance) be a workable option, here?

E.g. have the static `YourLog.Logger` property just set/read the `Log.Logger` property, and forward calls to it.

This would also prevent events being lost, in case code somewhere uses `Log.X()` inadvertently.

HTH!

**Comment by FaustoNascimento** [comment]
Umm... if I were to wrap the `Log` class, I'd have to do it "properly" - which means implementing all methods/properties that are available in the `Log` class as well as ensuring that instead of calling `Logger.<method>` that I called `Log.<method>` for abstraction reasons. Effectively the wrapper class would never know about an actual `ILogger` object directly, just what it wrapped around which is the `Log` class. That way it would ensure that if you decide to change anything in the `Log` class the wrapper - which would then behave as an actual wrapper - continued to work without any need for change in my code.

In other words, yes this is doable, but a lot more work than just copy/pasting the current `Log` class into my own and adding what I require that is missing for my needs. But that can't be done without `SilengLogger` being made public.

But more to the point: I think that having a `SilentLogger` class that we can easily and safely dump messages to is invaluable. I can't think of a reason why it can't be (safely) made public, but you obviously know the code and the direction of the project better than me :)

**Comment by FaustoNascimento** [comment]
Another reason for making the `SilentLogger` class public is the other ticket I opened [1210](https://github.com/serilog/serilog/issues/1210). 

A very simple extension method I use is this:

    public static ILogger When(this ILogger logger, Func<bool> predicate)
    {
        return predicate.Invoke() ? logger : null;
    }

Because I'm potentially returning `null` here, I have to invoke it as such:

    MyLogger.When(() => condition == true)?.Information(...)

If `SilentLogger` was made `public` I'd be able to ensure I always return an `ILogger` object.
